plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -799,7 +799,15 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
                         icebergTable.schema().findType(field.sourceId())))
                 .toArray(Type[]::new);
 
-        AppendFiles appendFiles = transaction.newAppend();
+        AppendFiles appendFiles;
+        if (session.getProperty("merge_manifests_on_write", Boolean.class)) {
+            log.info("Insert with merge_manifests_on_write = true -> Append()");
```

Comment:
Please remove this logging. We normally avoid logging like this during normal operation, since it's not typically useful to the administrator, and thus fills the logs and makes it harder to find relevant information about problems.

We could then simplify the code to be
```java
AppendFiles appendFiles = isMergeManifestsOnWrite(session) ? transaction.newAppend() : transaction.newFastAppend();

Reply:
Note that the Iceberg library `SnapshotProducer` already logs if the operation is `FastAppend` or `MergeAppend`.